### ISSUE TITLE
None should be checked with is None in docs

### ISSUE THREAD

**Issue by GeorgeFischhof** [tiangolo/fastapi#821]
Hi All

the library and the documentation are fantastic 

I am a QA eng., and teach Python for my colleagues.
I saw in some examples that None is checked as it would be a boolean expression
for example:
https://fastapi.tiangolo.com/tutorial/body/

async def create_item(item_id: int, item: Item, q: str = None):
    result = {"item_id": item_id, **item.dict()}
    if q:

I saw that the documentation is written for beginners too.
I think checking the None as it would be a boolean expression in the docs encourages beginners to use this formula instead of checking with is operator. And it is not good, because not None will them result unexpected results.

These examples should be corrected.

BR,
George

**Comment by shifr** [comment]
@GeorgeFischhof I wouldn't label it as a `bug` though. It's a valid expression and in the example `q` could be an empty string and we don't need it.

**Comment by GeorgeFischhof** [comment]
PR #824 added

**Comment by GeorgeFischhof** [comment]
Hi @shifr  yes, it is valid for strings, but singletons should be checked with is operator (as PEP8 says) and explicit is better than implicit ;-) And as you wrote too this check checks two thinks, but perhaps it is not the best for beginners

**Comment by shifr** [comment]
You're right about checking `is None` but in this case, we need to check if `q` exists and it is as explicit as needed. Also, don't forget about `Simple is better than complex.`

**Comment by tiangolo** [comment]
Thanks for the interest @GeorgeFischhof , but as I commented on the PR, I would prefer to keep it as is.

It's a bit more of a personal preference than best practices.

In fact, using `q is not None` would mean that `q` would be used even if it has an empty string `""`, which is probably not the intention. But thanks again for your time and interest! :rocket:
